File: script/make/protocol/js/GuildProtocol.js
export default class GuildProtocol {
    static encode(textEncoder, view, offset, protocol, data) {
        switch (protocol) {
            case 30101: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30102: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30103: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30104: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30105: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30106: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30107: {
                // extend
                while (view.byteLength < offset + 1) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                // 类型
                view.setUint8(offset, data["type"], false);
                offset = offset + 1;
                // extend
                while (view.byteLength < offset + 2) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                // 公会名
                const guildNameArray = textEncoder.encode(data["guildName"]);
                view.setUint16(offset, guildNameArray.length, false);
                offset = offset + 2;
                // extend
                while (view.byteLength < offset + guildNameArray.length) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                (new Uint8Array(view.buffer, offset)).set(guildNameArray);
                offset = offset + guildNameArray.length;
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30108: {
                // extend
                while (view.byteLength < offset + 8) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                // 公会ID
                view.setBigUint64(offset, data["guildId"], false);
                offset = offset + 8;
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30109: {
                // extend
                while (view.byteLength < offset + 8) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                // 公会ID
                view.setBigUint64(offset, data["guildId"], false);
                offset = offset + 8;
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30110: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30111: {
                // extend
                while (view.byteLength < offset + 8) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                // 角色ID
                view.setBigUint64(offset, data["roleId"], false);
                offset = offset + 8;
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30112: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30113: {
                // extend
                while (view.byteLength < offset + 8) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                // 角色ID
                view.setBigUint64(offset, data["roleId"], false);
                offset = offset + 8;
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30114: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30115: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30116: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30117: {
                // extend
                while (view.byteLength < offset + 8) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                // 角色ID
                view.setBigUint64(offset, data["roleId"], false);
                offset = offset + 8;
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30118: {
                // extend
                while (view.byteLength < offset + 8) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                // 角色ID
                view.setBigUint64(offset, data["roleId"], false);
                offset = offset + 8;
                // extend
                while (view.byteLength < offset + 1) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                // 位置
                view.setUint8(offset, data["job"], false);
                offset = offset + 1;
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30119: {
                return new DataView(view.buffer.slice(0, offset));
            }
            case 30120: {
                // extend
                while (view.byteLength < offset + 2) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                // 公告
                const noticeArray = textEncoder.encode(data["notice"]);
                view.setUint16(offset, noticeArray.length, false);
                offset = offset + 2;
                // extend
                while (view.byteLength < offset + noticeArray.length) {
                    const extendView = new DataView(new ArrayBuffer(view.byteLength * 2));
                    (new Uint8Array(extendView.buffer)).set(new Uint8Array(view.buffer));
                    view = extendView;
                }
                (new Uint8Array(view.buffer, offset)).set(noticeArray);
                offset = offset + noticeArray.length;
                return new DataView(view.buffer.slice(0, offset));
            }
            default: throw("unknown protocol define: " + protocol)
        }
    }

    static decode(textDecoder, view, offset, protocol) {
        switch (protocol) {
            case 30101: {
                // 公会列表
                const list = [];
                let listLength = view.getUint16(offset, false);
                offset = offset + 2;
                while (--listLength >= 0) {
                    // Guild
                    // 公会ID
                    const guildId = view.getBigUint64(offset, false);
                    offset = offset + 8;
                    // 公会名字
                    const guildNameLength = view.getUint16(offset, false);
                    offset = offset + 2;
                    const guildNameArray = new Uint8Array(view.buffer.slice(offset, offset + guildNameLength));
                    const guildName = textDecoder.decode(guildNameArray);
                    offset = offset + guildNameLength;
                    // 创建时间
                    const createTime = view.getUint32(offset, false);
                    offset = offset + 4;
                    // 会长角色ID
                    const leaderRoleId = view.getBigUint64(offset, false);
                    offset = offset + 8;
                    // 会长名字
                    const leaderNameLength = view.getUint16(offset, false);
                    offset = offset + 2;
                    const leaderNameArray = new Uint8Array(view.buffer.slice(offset, offset + leaderNameLength));
                    const leaderName = textDecoder.decode(leaderNameArray);
                    offset = offset + leaderNameLength;
                    // object
                    const guild = {guildId, guildName, createTime, leaderRoleId, leaderName};
                    // add
                    list.push(guild);
                }
                return {list};
            }
            case 30102: {
                // 成员列表
                const list = [];
                let listLength = view.getUint16(offset, false);
                offset = offset + 2;
                while (--listLength >= 0) {
                    // GuildRole
                    // 成员ID
                    const roleId = view.getBigUint64(offset, false);
                    offset = offset + 8;
                    // 职位
                    const job = view.getUint8(offset, false);
                    offset = offset + 1;
                    // 加入时间
                    const joinTime = view.getUint32(offset, false);
                    offset = offset + 4;
                    // 成员名字
                    const roleNameLength = view.getUint16(offset, false);
                    offset = offset + 2;
                    const roleNameArray = new Uint8Array(view.buffer.slice(offset, offset + roleNameLength));
                    const roleName = textDecoder.decode(roleNameArray);
                    offset = offset + roleNameLength;
                    // 性别
                    const sex = view.getUint8(offset, false);
                    offset = offset + 1;
                    // 职业
                    const classes = view.getUint8(offset, false);
                    offset = offset + 1;
                    // Vip等级
                    const vipLevel = view.getUint8(offset, false);
                    offset = offset + 1;
                    // object
                    const guildRole = {roleId, job, joinTime, roleName, sex, classes, vipLevel};
                    // add
                    list.push(guildRole);
                }
                return {list};
            }
            case 30103: {
                // 申请列表
                const list = [];
                let listLength = view.getUint16(offset, false);
                offset = offset + 2;
                while (--listLength >= 0) {
                    // GuildApply
                    // 申请ID
                    const roleId = view.getBigUint64(offset, false);
                    offset = offset + 8;
                    // 申请时间
                    const applyTime = view.getUint32(offset, false);
                    offset = offset + 4;
                    // 申请名字
                    const roleNameLength = view.getUint16(offset, false);
                    offset = offset + 2;
                    const roleNameArray = new Uint8Array(view.buffer.slice(offset, offset + roleNameLength));
                    const roleName = textDecoder.decode(roleNameArray);
                    offset = offset + roleNameLength;
                    // 性别
                    const sex = view.getUint8(offset, false);
                    offset = offset + 1;
                    // 职业
                    const classes = view.getUint8(offset, false);
                    offset = offset + 1;
                    // Vip等级
                    const vipLevel = view.getUint8(offset, false);
                    offset = offset + 1;
                    // object
                    const guildApply = {roleId, applyTime, roleName, sex, classes, vipLevel};
                    // add
                    list.push(guildApply);
                }
                return {list};
            }
            case 30104: {
                // Guild
                // 公会ID
                const guildId = view.getBigUint64(offset, false);
                offset = offset + 8;
                // 公会名字
                const guildNameLength = view.getUint16(offset, false);
                offset = offset + 2;
                const guildNameArray = new Uint8Array(view.buffer.slice(offset, offset + guildNameLength));
                const guildName = textDecoder.decode(guildNameArray);
                offset = offset + guildNameLength;
                // 经验
                const exp = view.getUint32(offset, false);
                offset = offset + 4;
                // 财富
                const wealth = view.getUint32(offset, false);
                offset = offset + 4;
                // 等级
                const level = view.getUint8(offset, false);
                offset = offset + 1;
                // 创建时间
                const createTime = view.getUint32(offset, false);
                offset = offset + 4;
                // 公告
                const noticeLength = view.getUint16(offset, false);
                offset = offset + 2;
                const noticeArray = new Uint8Array(view.buffer.slice(offset, offset + noticeLength));
                const notice = textDecoder.decode(noticeArray);
                offset = offset + noticeLength;
                // 会长角色ID
                const leaderRoleId = view.getBigUint64(offset, false);
                offset = offset + 8;
                // 会长名字
                const leaderNameLength = view.getUint16(offset, false);
                offset = offset + 2;
                const leaderNameArray = new Uint8Array(view.buffer.slice(offset, offset + leaderNameLength));
                const leaderName = textDecoder.decode(leaderNameArray);
                offset = offset + leaderNameLength;
                // object
                const guild = {guildId, guildName, exp, wealth, level, createTime, notice, leaderRoleId, leaderName};
                return {guild};
            }
            case 30105: {
                // GuildRole
                // 成员ID
                const roleId = view.getBigUint64(offset, false);
                offset = offset + 8;
                // 职位
                const job = view.getUint8(offset, false);
                offset = offset + 1;
                // 加入时间
                const joinTime = view.getUint32(offset, false);
                offset = offset + 4;
                // 成员名字
                const roleNameLength = view.getUint16(offset, false);
                offset = offset + 2;
                const roleNameArray = new Uint8Array(view.buffer.slice(offset, offset + roleNameLength));
                const roleName = textDecoder.decode(roleNameArray);
                offset = offset + roleNameLength;
                // 性别
                const sex = view.getUint8(offset, false);
                offset = offset + 1;
                // 职业
                const classes = view.getUint8(offset, false);
                offset = offset + 1;
                // Vip等级
                const vipLevel = view.getUint8(offset, false);
                offset = offset + 1;
                // object
                const guildRole = {roleId, job, joinTime, roleName, sex, classes, vipLevel};
                return {guildRole};
            }
            case 30106: {
                // 
                const list = [];
                let listLength = view.getUint16(offset, false);
                offset = offset + 2;
                while (--listLength >= 0) {
                    // GuildApply
                    // 公会ID
                    const guildId = view.getBigUint64(offset, false);
                    offset = offset + 8;
                    // 申请时间
                    const applyTime = view.getUint32(offset, false);
                    offset = offset + 4;
                    // 公会名字
                    const guildNameLength = view.getUint16(offset, false);
                    offset = offset + 2;
                    const guildNameArray = new Uint8Array(view.buffer.slice(offset, offset + guildNameLength));
                    const guildName = textDecoder.decode(guildNameArray);
                    offset = offset + guildNameLength;
                    // object
                    const guildApply = {guildId, applyTime, guildName};
                    // add
                    list.push(guildApply);
                }
                return {list};
            }
            case 30107: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30108: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30109: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30110: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30111: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30112: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30113: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30114: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30115: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30116: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30117: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30118: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30119: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            case 30120: {
                // 结果
                const resultLength = view.getUint16(offset, false);
                offset = offset + 2;
                const resultArray = new Uint8Array(view.buffer.slice(offset, offset + resultLength));
                const result = textDecoder.decode(resultArray);
                offset = offset + resultLength;
                return {result};
            }
            default: throw("unknown protocol define: " + protocol)
        }
    }
}

File: script/make/protocol/erl/guild_protocol.erl
-module(guild_protocol).
-export([decode/2, encode/2]).
-include("guild.hrl").

-spec decode(Protocol :: non_neg_integer(), Binary :: binary()) -> {ok, [integer() | binary() | list()]} | {error, Protocol :: non_neg_integer(), Binary :: binary()}.
decode(30101, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30102, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30103, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30104, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30105, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30106, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30107, _Rest_ = <<_/binary>>) ->
    <<Type:8, _TypeRest_/binary>> = _Rest_,
    <<GuildNameLength:16, GuildName:GuildNameLength/binary, _GuildNameRest_/binary>> = _TypeRest_,
    {ok, {Type, GuildName}};

decode(30108, _Rest_ = <<_/binary>>) ->
    <<:64, _Rest_/binary>> = _Rest_,
    {ok, };

decode(30109, _Rest_ = <<_/binary>>) ->
    <<:64, _Rest_/binary>> = _Rest_,
    {ok, };

decode(30110, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30111, _Rest_ = <<_/binary>>) ->
    <<:64, _Rest_/binary>> = _Rest_,
    {ok, };

decode(30112, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30113, _Rest_ = <<_/binary>>) ->
    <<:64, _Rest_/binary>> = _Rest_,
    {ok, };

decode(30114, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30115, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30116, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30117, _Rest_ = <<_/binary>>) ->
    <<:64, _Rest_/binary>> = _Rest_,
    {ok, };

decode(30118, _Rest_ = <<_/binary>>) ->
    <<RoleId:64, _RoleIdRest_/binary>> = _Rest_,
    <<Job:8, _JobRest_/binary>> = _RoleIdRest_,
    {ok, {RoleId, Job}};

decode(30119, _Rest_ = <<_/binary>>) ->
    {ok, {}};

decode(30120, _Rest_ = <<_/binary>>) ->
    <<Length:16, :Length/binary, _Rest_/binary>> = _Rest_,
    {ok, };

decode(Protocol, Binary) ->
    {error, Protocol, Binary}.


-spec encode(Protocol :: non_neg_integer(), Data :: atom() | tuple() | binary() | list()) -> {ok, binary()} | {error, Protocol :: non_neg_integer(), Data :: atom() | tuple() | binary() | list()}.
encode(30101, ) ->
    Data30101 = <<(encode__30101(<<>>, 0, ets:safe_fixtable(, true) andalso , ets:first()))/binary>>,
    {ok, <<(byte_size(Data30101)):16, 30101:16, Data30101/binary>>};

encode(30102, ) ->
    Data30102 = <<(encode__30102(<<>>, 0, ets:safe_fixtable(, true) andalso , ets:first()))/binary>>,
    {ok, <<(byte_size(Data30102)):16, 30102:16, Data30102/binary>>};

encode(30103, ) ->
    Data30103 = <<(encode__30103(<<>>, 0, ets:safe_fixtable(, true) andalso , ets:first()))/binary>>,
    {ok, <<(byte_size(Data30103)):16, 30103:16, Data30103/binary>>};

encode(30104, #guild{guild_id = GuildId, guild_name = GuildName, exp = Exp, wealth = Wealth, level = Level, create_time = CreateTime, notice = Notice, leader_role_id = LeaderRoleId, leader_name = LeaderName}) ->
    Data30104 = <<GuildId:64, (byte_size(GuildName)):16, (GuildName)/binary, Exp:32, Wealth:32, Level:8, CreateTime:32, (byte_size(Notice)):16, (Notice)/binary, LeaderRoleId:64, (byte_size(LeaderName)):16, (LeaderName)/binary>>,
    {ok, <<(byte_size(Data30104)):16, 30104:16, Data30104/binary>>};

encode(30105, #guild_role{role_id = RoleId, job = Job, join_time = JoinTime, role_name = RoleName, sex = Sex, classes = Classes, vip_level = VipLevel}) ->
    Data30105 = <<RoleId:64, Job:8, JoinTime:32, (byte_size(RoleName)):16, (RoleName)/binary, Sex:8, Classes:8, VipLevel:8>>,
    {ok, <<(byte_size(Data30105)):16, 30105:16, Data30105/binary>>};

encode(30106, ) ->
    Data30106 = <<(encode__30106(<<>>, 0, ))/binary>>,
    {ok, <<(byte_size(Data30106)):16, 30106:16, Data30106/binary>>};

encode(30107, ) ->
    Data30107 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30107)):16, 30107:16, Data30107/binary>>};

encode(30108, ) ->
    Data30108 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30108)):16, 30108:16, Data30108/binary>>};

encode(30109, ) ->
    Data30109 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30109)):16, 30109:16, Data30109/binary>>};

encode(30110, ) ->
    Data30110 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30110)):16, 30110:16, Data30110/binary>>};

encode(30111, ) ->
    Data30111 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30111)):16, 30111:16, Data30111/binary>>};

encode(30112, ) ->
    Data30112 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30112)):16, 30112:16, Data30112/binary>>};

encode(30113, ) ->
    Data30113 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30113)):16, 30113:16, Data30113/binary>>};

encode(30114, ) ->
    Data30114 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30114)):16, 30114:16, Data30114/binary>>};

encode(30115, ) ->
    Data30115 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30115)):16, 30115:16, Data30115/binary>>};

encode(30116, ) ->
    Data30116 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30116)):16, 30116:16, Data30116/binary>>};

encode(30117, ) ->
    Data30117 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30117)):16, 30117:16, Data30117/binary>>};

encode(30118, ) ->
    Data30118 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30118)):16, 30118:16, Data30118/binary>>};

encode(30119, ) ->
    Data30119 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30119)):16, 30119:16, Data30119/binary>>};

encode(30120, ) ->
    Data30120 = <<(protocol:text())/binary>>,
    {ok, <<(byte_size(Data30120)):16, 30120:16, Data30120/binary>>};

encode(Protocol, Data) ->
    {error, Protocol, Data}.

encode__30101(Acc = <<_/binary>>, Length, Tab, '$end_of_table') ->
    ets:safe_fixtable(Tab, false),
    <<Length:16, Acc/binary>>;
encode__30101(Acc = <<_/binary>>, Length, Tab, Key) ->
    case ets:lookup(Tab, Key) of
        [] ->
            encode__30101(Acc, Length, Tab, ets:next(Tab, Key));
        [#guild{guild_id = GuildId, guild_name = GuildName, create_time = CreateTime, leader_role_id = LeaderRoleId, leader_name = LeaderName}] ->
            encode__30101(<<Acc/binary, GuildId:64, (byte_size(GuildName)):16, (GuildName)/binary, CreateTime:32, LeaderRoleId:64, (byte_size(LeaderName)):16, (LeaderName)/binary>>, Length + 1, Tab, ets:next(Tab, Key))
    end.

encode__30102(Acc = <<_/binary>>, Length, Tab, '$end_of_table') ->
    ets:safe_fixtable(Tab, false),
    <<Length:16, Acc/binary>>;
encode__30102(Acc = <<_/binary>>, Length, Tab, Key) ->
    case ets:lookup(Tab, Key) of
        [] ->
            encode__30102(Acc, Length, Tab, ets:next(Tab, Key));
        [#guild_role{role_id = RoleId, job = Job, join_time = JoinTime, role_name = RoleName, sex = Sex, classes = Classes, vip_level = VipLevel}] ->
            encode__30102(<<Acc/binary, RoleId:64, Job:8, JoinTime:32, (byte_size(RoleName)):16, (RoleName)/binary, Sex:8, Classes:8, VipLevel:8>>, Length + 1, Tab, ets:next(Tab, Key))
    end.

encode__30103(Acc = <<_/binary>>, Length, Tab, '$end_of_table') ->
    ets:safe_fixtable(Tab, false),
    <<Length:16, Acc/binary>>;
encode__30103(Acc = <<_/binary>>, Length, Tab, Key) ->
    case ets:lookup(Tab, Key) of
        [] ->
            encode__30103(Acc, Length, Tab, ets:next(Tab, Key));
        [#guild_apply{role_id = RoleId, apply_time = ApplyTime, role_name = RoleName, sex = Sex, classes = Classes, vip_level = VipLevel}] ->
            encode__30103(<<Acc/binary, RoleId:64, ApplyTime:32, (byte_size(RoleName)):16, (RoleName)/binary, Sex:8, Classes:8, VipLevel:8>>, Length + 1, Tab, ets:next(Tab, Key))
    end.

encode__30106(Acc = <<_/binary>>, Length, []) ->
    <<Length:16, Acc/binary>>;
encode__30106(Acc = <<_/binary>>, Length, [#guild_apply{guild_id = GuildId, apply_time = ApplyTime, guild_name = GuildName} | ]) ->
    encode__30106(<<Acc/binary, GuildId:64, ApplyTime:32, (byte_size(GuildName)):16, (GuildName)/binary>>, Length + 1, ).

